tests.py

```diff
@@ -124,198 +129,227 @@ def test_current_dir_config(self):
 
 
 class RequestTest(TestCase):
-    @patch('requests.get')
-    def test_request_params(self, get):
-        cs = CloudStack(endpoint='localhost', key='foo', secret='bar',
+    @patch("requests.Session.send")
+    def test_request_params(self, mock):
+        cs = CloudStack(endpoint='https://localhost', key='foo', secret='bar',
                         timeout=20, expiration=-1)
-        get.return_value.status_code = 200
-        get.return_value.json.return_value = {
+        mock.return_value.status_code = 200
+        mock.return_value.json.return_value = {
             'listvirtualmachinesresponse': {},
         }
         machines = cs.listVirtualMachines(listall='true',
                                           headers={'Accept-Encoding': 'br'})
         self.assertEqual(machines, {})
-        get.assert_called_once_with(
-            'localhost', timeout=20, verify=True, cert=None,
-            headers={
-                'Accept-Encoding': 'br',
-            },
-            params={
-                'apiKey': 'foo',
-                'response': 'json',
-                'command': 'listVirtualMachines',
-                'listall': 'true',
-                'signature': 'B0d6hBsZTcFVCiioSxzwKA9Pke8=',
-            },
-        )
-
-    @patch('requests.get')
-    def test_request_params_casing(self, get):
-        cs = CloudStack(endpoint='localhost', key='foo', secret='bar',
+
+        mock.assert_called_once()
+
+        args, kwargs = mock.call_args
```

Comment:
If you don't need args:

```suggestion
        [request], kwargs = mock.call_args
```